### PR TITLE
created a composite listener for preview #111

### PR DESCRIPTION
**Pull request by Earthii**

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt [hunk 1]
```diff
@@ -93,18 +89,24 @@ class Preview : ViewGroup, SurfaceHolder.Callback, MediaScannerConnection.OnScan
...
         initGestureDetector()
 
+        var groupListener = CompositeOnTouchListener()
 
-        mSurfaceView.setOnTouchListener { view, event ->
-            mLastClickX = event.x.toInt()
-            mLastClickY = event.y.toInt()
+        groupListener.addOnTouchListener(object:OnTouchListener{
+            override fun onTouch(v: View?, event: MotionEvent?): Boolean {
+                Log.d("focus", "In focus")
```
Comment (n2chao) [inline app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt]: Remove unnecessary log

File: app/src/main/kotlin/com/simplemobiletools/camera/helpers/OnSwipeTouchListener.kt [hunk 2]
```diff
@@ -19,6 +20,7 @@ open class OnSwipeTouchListener(ctx: Context) : OnTouchListener {
     }
 
     override fun onTouch(v: View, event: MotionEvent): Boolean {
+        Log.d("Swipe", "In Swipe")
```
Comment (n2chao) [inline app/src/main/kotlin/com/simplemobiletools/camera/helpers/OnSwipeTouchListener.kt]: Remove unnecessary log

File: app/src/main/kotlin/com/simplemobiletools/camera/helpers/CompositeOnTouchListener.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import android.view.MotionEvent
+import android.view.View
+
+/**
+ * Created by ericxiao on 2018-03-23.
+ */
+class CompositeOnTouchListener : View.OnTouchListener {
+
+    override fun onTouch(v: View?, event: MotionEvent?): Boolean {
+        for (listener in listeners) {
```
Comment (StevenHaa) [inline app/src/main/kotlin/com/simplemobiletools/camera/helpers/CompositeOnTouchListener.kt]: is there a reason why the variable "listeners" is used here before it was called on line 19?

File: app/src/main/kotlin/com/simplemobiletools/camera/helpers/OnSwipeTouchListener.kt [hunk 4]
```diff
@@ -4,6 +4,7 @@ package com.simplemobiletools.camera.helpers
  * Created by ericxiao on 2018-03-16.
  */
 import android.content.Context
+import android.util.Log
```
Comment (StevenHaa) [inline app/src/main/kotlin/com/simplemobiletools/camera/helpers/OnSwipeTouchListener.kt]: should we also remove the log import?

File: app/src/test/java/Test_OnSwipeTouchListener.kt [hunk 5]
```diff
@@ -27,10 +28,8 @@ class Test_OnSwipeTouchListener {
 
     private lateinit var activity: MainActivity
     private lateinit var surfaceView : SurfaceView
-    val mockCamera = Mockito.mock(Camera::class.java)
-    val mockParameters = Mockito.mock(Camera.Parameters::class.java)
     private lateinit var testPreview : Preview
-    var myListener : OnSwipeTouchListener? = null
+    var myListener : CompositeOnTouchListener? = null
```
Comment (StevenHaa) [inline app/src/test/java/Test_OnSwipeTouchListener.kt]: can you change myListener to compositeOnTouchListener?

File: app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt (hunk 1 above)
Comment (Earthii) [reply app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt]: done

**Comment by Ea10** [comment]
#111

**Pull request merged**